compiler/rustc_middle/src/dep_graph/mod.rs

```diff
@@ -50,6 +56,12 @@
 //! than a zeroed out fingerprint. More generally speaking, it relieves the
```

Comment:
I am not sure the API is that restricted any more. There is an escape hatch `DepNode::new_no_params` which is used in non-incremental runs.